ruby-sdk/ext/eppo_client/Cargo.toml

```diff
@@ -16,6 +16,7 @@ eppo_core = { version = "4.0.0" }
 log = { version = "0.4.21", features = ["kv_serde"] }
 magnus = { version = "0.6.4" }
 serde = { version = "1.0.203", features = ["derive"] }
+openssl = { version = "*", features = ["vendored"] }
```

Comment:
I believe we don't need to vendor openssl on macOS and Windows because reqwest will use system's native library.

You can take a look at python SDK: https://github.com/Eppo-exp/rust-sdk/blob/427fed8fd54f3bdc7b69a9f2e198a72ef5ba4779/python-sdk/Cargo.toml#L18-L36

This should also fix macOS build.